pandas/_libs/tslibs/strptime.pyx

```diff
@@ -232,9 +237,21 @@ cdef _get_format_regex(str fmt):
 
 
 cdef class DatetimeParseState:
-    def __cinit__(self):
+    def __cinit__(self, NPY_DATETIMEUNIT creso=NPY_DATETIMEUNIT.NPY_FR_ns):
         self.found_tz = False
         self.found_naive = False
+        self.creso = creso
+        self.creso_changed = False
+
+    cdef bint update_creso(self, NPY_DATETIMEUNIT item_reso) noexcept:
+        # Return a bool indicating whether we bumped to a higher resolution
+        if self.creso == NPY_DATETIMEUNIT.NPY_FR_GENERIC:
+            self.creso = item_reso
+        elif item_reso > self.creso:
+            self.creso = item_reso
+            self.creso_changed = True
```

Comment:
So we only want to track if this changes once from the default? e.g.

```
>>> s = DatetimeParseState("s")
>>> s.update_creso("ms")
>>> s.update_creso("ms")
>>> s.creso_changed  # should still return True?
```

Reply:
correct.  this is a has-it-ever-changed-since-init

Reply:
Gotcha. Could we rename this like `creso_changed_from_default`?

Reply:
sure, will update

Reply:
updated